### ISSUE TITLE
Layout to integrate Dockerfile for multiple distributions

### ISSUE THREAD

**Issue by jcfr** [thewtex/SlicerDocker#6]
Currently the Dockerfile are based on centos5, this is great and should not change.

In addition of these default docker files, I would like store Dockerfile that would allow to easily test the build of Slicer on other distribution, it would be a systematic way of documenting and testing what is already on the Slicer wiki [here](http://wiki.slicer.org/slicerWiki/index.php/Documentation/Nightly/Developers/Build_Instructions)

I propose the following layout:

```
<root>
  |- slicer-build-base
  |           |- Dockerfile   # Default: centos5
  |           |
  |           |- Ubuntu-16.04-amd64
  |           |       |- DockerFile
  |           |
  |           |- Debian-8.4.0-amd64
  |                   |- DockerFile
  |
  |- slicer-build-deps
  |           |- Dockerfile   # Default: centos5
  |           |
  |           |- Ubuntu-16.04-amd64
  |           |       |- DockerFile
  |           |
  |           |- Debian-8.4.0-amd64
  |                   |- DockerFile
  |
  |- slicer-build
  |           |- Dockerfile   # Default: centos5
  |           |
  |           |- Ubuntu-16.04
  |           |       |- DockerFile
  |           |
  |           |- Debian-8.4.0-amd64
  |                   |- DockerFile
```

with the introduction of a new intermediate image stored on [commontk](https://github.com/commontk/CTK)/CTKDocker:
```
<root>
  |- qt-build
  |           |- Dockerfile   # Default: centos5
  |           |
  |           |- Ubuntu-16.04
  |           |       |- DockerFile
  |           |
  |           |- Debian-8.4.0-amd64
  |                   |- DockerFile
```

**Comment by jcfr** [comment]
Cc: @thewtex @QuadmasterXLII

**Comment by jcfr** [comment]
For reference: https://docs.docker.com/engine/reference/commandline/build/

**Comment by thewtex** [comment]
@jcfr sounds good.

After first getting the images up, we could consider finding a way to prevent duplicate content in the Dockerfile's.  Or, we may find that the complexity is not worth it.

**Comment by jcfr** [comment]
> we may find that the complexity is not worth it.

Agreed. This is a concern. It is a balance between complexity and simplicity.

> prevent duplicate content

to avoid duplicated content, we could use dockerfile generator ... https://www.npmjs.com/package/dockerfile-generator

Related reads:
* http://jrruethe.github.io/blog/2015/09/20/dockerfile-generator/


I suggest we wait to have few of the Dockerfile before coming with a solution avoiding duplication ..
